controllers/volumereplication_controller.go

```diff
@@ -162,28 +171,42 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case replicationv1alpha1.Primary:
 		err := r.markVolumeAsPrimary(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Secondary:
 		err := r.markVolumeAsSecondary(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Resync:
 		err := r.resyncVolume(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	default:
+		r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, ("unsupported image state " + string(instance.Spec.ImageState)))
 		return ctrl.Result{}, fmt.Errorf("unsupported image state %q", instance.Spec.ImageState)
 	}
 
+	instance.Status.LastCompletionTime = metav1.Now()
+	r.updateStatus(instance, replicationv1alpha1.Replicating, "volume replication is ready to use")
```

Comment:
let's remove `volume replication is ready to use now. keep the message as `Volume is marked primary or {secondary}` in case of resync `volume is marked for resyncing  

@ShyamsundarR any suggestion here?

Reply:
Done